src/routes/share.js

```diff
@@ -0,0 +1,122 @@
+import express from 'express';
+let router = express.Router();
+import knex from "../knex";
+
+router.post('/', (req, res) => newShareWrapper(req, res));
+router.get('/:uuid', (req, res) => getCorrespondingLinkWrapper(req, res));
+
+
+// Description: newShareMapper generates a new uuid which will be the unique link (or reuses the already existing one)
+// body: {
+//   "provider": string,
+//   "hashedUrl": string
+//   "date": string, // date needs to be of this format: yyyy-mm-dd hh-mm-ss.nnnnnn(+/- offset to utc time)
+//   "city": string
+// }
+// response: {
+//   success: true //or false if something went wrong, most likely the body wasn't correct
+//   "shareId": ...
+// }
+//
+const newShareWrapper = async (req, res) => {
+    //Generating uuid for unique link
+    const uuid = crypto.randomUUID();
+    try {
+        //Use this code as well, if you don't want to generate a new link if one already exists
+        /*const doesExist = await knex('disposible')
+            .select('link')
+            .whereRaw('LOWER(provider) = LOWER(?)', req.body.provider)
+            .whereRaw('LOWER(hashed_url) = LOWER(?)', req.body.hashedUrl);
+        if (typeof doesExist[0] !== 'undefined') {
+            console.log(doesExist[0]);
+            res.status(200).json({ success: true, shareId: doesExist[0].link });
+            return;
+        }*/
+
+        //Inserting new share link
+        const result = await knex('disposible')
+            .insert({
+                provider: req.body.provider,
+                hashed_url: req.body.hashedUrl,
+                link: uuid,
+                calendar_date: req.body.date,
+                city_slug: req.body.city
+        })
+            .then((result) => {
+                res.status(200).json({ success: true, shareId: uuid });
+            })
+    } catch (error) {
+        console.log('Error inserting link. ' + error);
+        res.status(500).json({ success: false, error: 'Failed to generate link.' });
+    }
+
+}
+
+// Description: Retrieves data for a sharing link
+// Optional query param: city //this is the city saved by the cookie of the user that opens the link
+// response: {
+//   "success": true //or false if something went wrong, most likely the body wasn't correct
+//   "date": ...,
+//   "city": ...,
+//   "tourId": ..., //current temporary id of the tour
+//   "usedCityOfCookie" //Whether the city of the cookie is returned or the one of when the link was generated
+// }
+//
+const getCorrespondingLinkWrapper = async (req, res) => {
+    //Share id from query param
+    const shareId = req.params.uuid;
+    //City of Cookie, when there is none it will be set to null
+    let citySlugOfCookie = req.query && req.query.city ? req.query.city : null;
+    //Variable that changes depending on whether the city of the cookie has been used or the one of the link creator
+    let usedCityOfCookie = true;
```

Comment:
It would be more intuitive to start with the value `false`, since we have yet to find out, if we will use the city of the cookie - the conditions have to be turned around then and we can set the value to `true`, when we know we can use this city.